howto/monitor/collect-metrics.md

```diff
@@ -0,0 +1,227 @@
+amd64          | 4         | 4 GB | 100 GB SSD | no   | Grafana, Prometheus |
+
+Complete the following steps to deploy Anbox Cloud with the reference monitoring stack:
+
+1. Create a `monitoring.yaml` file with the following content, depending on which [Juju bundle](https://discourse.ubuntu.com/t/about-anbox-cloud/17802#juju-bundles) you want to deploy:
+
+   [Details="For the `anbox-cloud-core` bundle"]
+   ```yaml
+   applications:
+     grafana-proxy:
+       charm: 'cs:haproxy'
+       num_units: 1
+       expose: true
+       options:
+         default_mode: tcp
+         peering_mode: active-active
+         ssl_cert: SELFSIGNED
+         ssl_key: SELFSIGNED
+         services: |
+           - service_name: app-grafana
+             service_host: "0.0.0.0"
+             service_port: 443
+             service_options:
+               - mode http
+               - http-request set-path %[path,regsub(^/grafana/?,/)]
+               - balance leastconn
+             server_options: check verify none inter 2000 rise 2 fall 5 maxconn 4096
+             crts: [DEFAULT]
+           - service_name: http_service
+             service_host: "0.0.0.0"
+             service_port: 80
+             service_options:
+               - mode http
+               - http-request redirect scheme https
+       to:
+         - '2'
+
+     prometheus:
+       charm: 'cs:prometheus2-18'
+       num_units: 1
+       to:
+         - '2'
+
+     ams-monitor:
+       charm: 'cs:telegraf-39'
+       options:
+         tags: region=cloud-0
+         prometheus_output_port: "20003"
+
+     lxd-monitor:
+       charm: 'cs:telegraf-39'
+       options:
+         tags: region=cloud-0
+         prometheus_output_port: "20004"
+
+     grafana:
+       charm: 'cs:grafana-36'
+       num_units: 1
+       options:
+         root_url: '%(protocol)s://%(domain)s:%(http_port)s/grafana'
+       to:
+         - '2'
+
+   relations:
+     - ['ams:prometheus', 'prometheus:scrape']
+     - ['grafana:grafana-source', 'prometheus:grafana-source']
+     - ['grafana-proxy:reverseproxy', 'grafana:website']
+     - ['lxd-monitor:prometheus-client', 'prometheus:target']
+     - ['lxd-monitor:juju-info', 'lxd:juju-info']
+     - ['ams-monitor:prometheus-client', 'prometheus:target']
+     - ['ams-monitor:juju-info', 'ams:juju-info']
+     - ['ams:grafana', 'grafana:dashboards']
+
+   machines:
+     '0':
+       series: focal
+       constraints: "cpu-cores=4 mem=8G root-disk=100G"
+     '1':
+       series: focal
+       constraints: "cpu-cores=8 mem=16G root-disk=200G"
+     '2':
+       series: focal
+       constraints: "cpu-cores=4 mem=4G root-disk=100G"
+   ```
+   [/Details]
+
+   [Details="For the `anbox-cloud` bundle"]
+   ```yaml
+   applications:
+     grafana-proxy:
+       charm: 'cs:haproxy'
+       num_units: 1
+       expose: true
+       options:
+         default_mode: tcp
+         peering_mode: active-active
+         ssl_cert: SELFSIGNED
+         ssl_key: SELFSIGNED
```

Comment:
Lets add a comment that these should be replaced with proper cert and key rather than using auto generated and self signed ones.

Reply:
... for a production-level solution, right?

Reply:
correct